### PR TITLE
FEATURE: FET-203 remove cyclic dependencies part 1

### PR DESCRIPTION
**Pull request by tdtom167**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-app-components:
- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4096/ : SUCCESS in 12m 38s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5213/ : SUCCESS in 10s

File: src/components/visualizations/chart/colorFactory.ts [hunk 1]
```diff
@@ -117,6 +117,25 @@ export abstract class ColorStrategy implements IColorStrategy {
 
 const emptyColorPaletteItem: IGuidColorItem = { type: "guid", value: "none" };
 
+export function findParentMeasureIndex(afm: AFM.IAfm, measureItemIndex: number): number {
```
Comment (martinmosko) [inline src/components/visualizations/chart/colorFactory.ts]: belongs this into module context?

File: src/components/visualizations/chart/colorFactory.ts [hunk 2]
```diff
@@ -587,6 +606,18 @@ export function isAttributeColorPalette(type: string, afm: AFM.IAfm, stackByAttr
     return stackByAttribute || (attributeChartSupported && afm.attributes && afm.attributes.length > 0);
 }
 
+export function isDerivedMeasure(measureItem: Execution.IMeasureHeaderItem, afm: AFM.IAfm) {
```
Comment (martinmosko) [inline src/components/visualizations/chart/colorFactory.ts]: belongs this into module context?

File: src/components/visualizations/chart/highcharts/commonConfiguration.ts [hunk 3]
```diff
@@ -8,14 +8,11 @@ import { css } from "highcharts";
 import { chartClick } from "../../utils/drilldownEventing";
 import { styleVariables } from "../../styles/variables";
 import { isOneOfTypes } from "../../utils/common";
-import { supportedDualAxesChartTypes } from "../chartOptionsBuilder";
+import { supportedDualAxesChartTypes, DEFAULT_CATEGORIES_LIMIT } from "../chartOptionsBuilder";
 import { setupDrilldown } from "../events/setupDrilldownToParentAttribute";
 
 const isTouchDevice = "ontouchstart" in window || navigator.msMaxTouchPoints;
 
-export const DEFAULT_SERIES_LIMIT = 1000;
```
Comment (martinmosko) [inline src/components/visualizations/chart/highcharts/commonConfiguration.ts]: I'm not sure if its a good idea to move constants from here to chart builder

File: src/components/visualizations/table/constants/tableConstants.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+// (C) 2007-2019 GoodData Corporation
```
Comment (martinmosko) [inline src/components/visualizations/table/constants/tableConstants.tsx]: if its a new file it should be just 2019

File: src/components/visualizations/table/utils/utils.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+// (C) 2019-2019 GoodData Corporation
```
Comment (martinmosko) [inline src/components/visualizations/table/utils/utils.ts]: just 2019 once

File: src/components/visualizations/table/utils/test/utils.spec.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
+// (C) 2007-2019 GoodData Corporation
+import { ITotalWithData } from "../../../../../interfaces/Totals";
+import { getFooterHeight, getHeaderOffset } from "../utils";
+describe("Table utils - Utils", () => {
```
Comment (martinmosko) [inline src/components/visualizations/table/utils/test/utils.spec.ts]: sounds to me like too common name for file... basically it does not say anything what is here... which is wrong

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4167/ : FAILURE in 6m 34s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5303/ : SUCCESS in 12s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4171/ : FAILURE in 2m 56s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5307/ : SUCCESS in 10s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4176/ : SUCCESS in 9m 12s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5314/ : SUCCESS in 13s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4177/ : FAILURE in 15m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5315/ : SUCCESS in 49s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4180/ : SUCCESS in 31m 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5318/ : SUCCESS in 11s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4193/ : FAILURE in 11m 14s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5338/ : SUCCESS in 13s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check-teardown pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-stop-zuul-docker/618/ : FAILURE in 10m 52s

**Pull request closed without merging**